lib/screens/search/search_results_page.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:my_app/graphql/client.dart';
+import 'package:my_app/graphql/queries/search.graphql.dart';
+import 'package:my_app/screens/search/models/search_result_item.dart';
+
+import 'episode_list.dart';
+import 'result_programs_list.dart';
+
+class SearchResultsPage extends ConsumerStatefulWidget {
+  final String searchInput;
+
+  const SearchResultsPage(this.searchInput);
+
+  @override
+  ConsumerState<SearchResultsPage> createState() => _SearchResultsPageState();
+}
+
+class _SearchResultsPageState extends ConsumerState<SearchResultsPage> {
+  var results = true;
+  late Future<Query$Search$search?> resultFuture;
+
+  @override
+  void didUpdateWidget(SearchResultsPage oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    final client = ref.read(gqlClientProvider);
+
+    if (widget.searchInput != '') {
+      resultFuture = client
+          .query$Search(
+        Options$Query$Search(
+          variables: Variables$Query$Search(queryString: widget.searchInput),
+        ),
+      )
+          .onError((error, stackTrace) {
+        1;
+        throw Error();
+      }).then(
+        (value) {
+          if (value.hasException) {
+            throw ErrorDescription(value.exception.toString());
+          }
+          return value.parsedData?.search;
+        },
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.searchInput == '') {
+      return noInputInfoWidget;
+    } else {
+      return FutureBuilder<Query$Search$search?>(
+        future: resultFuture,
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.waiting) {
+            return const Center(
+              child: SizedBox.square(
+                dimension: 50,
+                child: CircularProgressIndicator(),
+              ),
+            );
+          }
+          if (snapshot.hasData) {
+            var searchResults = snapshot.data!.result;
+            if (searchResults.isEmpty) {
+              return noResultsInfoWidget;
+            } else {
+              var results = getResultProgramsEpisodes(searchResults);
+              List<SearchResultItem> programs =
+                  results['programs'] as List<SearchResultItem>;
+              List<SearchResultItem> episodes =
+                  results['episodes'] as List<SearchResultItem>;
+              return ListView(
+                children: [
+                  if (programs.isNotEmpty)
+                    ResultProgramsList(title: 'Programs', items: programs),
+                  if (episodes.isNotEmpty)
+                    EpisodeList(title: 'Episodes', items: episodes),
```

Comment:
```suggestion
                    EpisodeList(title: 'Episodes', items: results.where((i) => i is Fragment$SearchResultItem$$EpisodeSearchItem).toList()),
```

Skip the getResultProgramsEpisodes and use the graphql types directly, like this :)

Reply:
Only the fragment class for EpisodeSearchItem got generated `Fragment$SearchResultItem$$EpisodeSearchItem` . There is no similar classes for Season and Show. So I would have to do a string comparison on  `__typename` ?